apps/sms/views/shared/js/settings.js

```diff
@@ -67,9 +70,37 @@ var Settings = {
       }
     }
 
+    //create lock for SEND_READ_REPORT_KEY
+    keyHandlerSet[this.SEND_READ_REPORT_KEY] = this.toSendReadReport.bind(this);
+
     for (var key in keyHandlerSet) {
       setHandlerMap(key);
     }
+
+    //add Observer for SEND_READ_REPORT_KEY
+    settings.addObserver(this.SEND_READ_REPORT_KEY, function (event) {
+      Settings.sendReadReportStatus = event.settingValue;
+      console.log(Settings.sendReadReportStatus);
+    });
+  },
+
+  sendReadReport: function toSendReadReport() {
+    var deferred = Utils.Promise.defer();
+    if(Settings.sendReadReportStatus != null) {
+      deferred.resolve(Settings.sendReadReportStatus);
+    } else {
+      deferred.reject('ERROR: Settings.sendReadReportStatus unavailable');
+    };
+
+    return deferred.promise;
+  },
+
+  //Set sendReadReportStatus for SEND_READ_REPORT_KEY
+  toSendReadReport: function toSendReadReport(status) {
+    if (!isNaN(status)) {
+      Settings.sendReadReportStatus = status;
+      console.log(Settings.sendReadReportStatus);
+    }
   },
```

Comment:
so it's not exactly what I had in mind, but it's much closer.

What I propose is that `sendReadReportStatus` is already a Promise (and we can call it simply `sendReadReport`).

So in init:
```javascript
... keyHandlerSet + init things

this._sendReadReportDefer = Utils.Promise.defer();
this.sendReadReport = this._sendReadReportDefer.promise;
```

in initSendReadReport:
```javascript
function initSendReadReport(value) {
  this._sendReadReportDefer.resolve(value);
  ...addObserver(settingName, (e) => this.sendReadReport = Promise.resolve(e.settingValue));
}
```

Tell me if this makes sense ?